### DISCUSSION TITLE
How to use IDS Messaging in EDC?

### DISCUSSION THREAD

**Discussion by mark-olchanski** [eclipse-edc/Connector#1402] [Q&A]
Hello,

we would like to implement extra layers of security when someone from the outside communicates with us and we figured that using IDS messaging would be a good way to go. Now we wonder if there is a way or endpoint or something else to easily use this within EDC?

We found that in the process of a contract negotiation and during the data transfer process the ids-multipart protocol is used, but are there any other places where this is implemented?

**Comment by mspiekermann** [comment]
The ids-multipart protocol in general is a deliverable of the International Dataspace Association (IDSA) and therefore discussed and developed in its working groups (mainly the architecture one). EDC uses this protocol in terms of adding the corresponding IDS extensions.

> Now we wonder if there is a way or endpoint or something else to easily use this within EDC?

The EDC supports the IDS multipart protocol as you have already found out by looking at the process of the policy negotiation. IDS-based data transfer is also part of the [samples](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/tree/aac26a8106a8f241b1dfa39c8d51c731d3dc0fa2/samples) that can be found in the EDC repository. They will allow you to gain more insights on how to use EDC with IDS.
